test/segment/analytics_test.exs

```diff
@@ -8,170 +8,125 @@ defmodule Segment.Analytics.AnalyticsTest do
+            "app" => nil,
+            "ip" => nil,
+            "library" => %{
+              "name" => "analytics_elixir",
+              "transport" => "http",
+              "version" => version
+            },
+            "location" => nil,
+            "os" => nil,
+            "page" => nil,
+            "referrer" => nil,
+            "screen" => nil,
+            "timezone" => nil,
+            "traits" => nil,
+            "userAgent" => nil
+          },
+          "event" => "test1",
+          "properties" => %{},
+          "timestamp" => nil,
+          "type" => "track",
+          "userId" => nil,
+          "version" => nil
+        }
+      ]
+    }
+
+    expected_response = ~s({"another": {"json": ["response"]}})
+
+    {:ok,
+     bypass: bypass,
+     event: event,
+     expected_request_body: expected_request_body,
+     expected_response: expected_response}
   end
 
   describe "call/2" do
-    test "sends an event", %{bypass: bypass, version: version} do
+    test "sends an event, and returns the response", %{
+      bypass: bypass,
+      event: event,
+      expected_request_body: expected_request_body,
+      expected_response: expected_response
+    } do
       Bypass.expect(bypass, fn conn ->
         {:ok, received_body, _conn} = Plug.Conn.read_body(conn)
 
-        assert %{
-                 "batch" => [
-                   %{
-                     "anonymousId" => nil,
-                     "context" => %{
-                       "app" => nil,
-                       "ip" => nil,
-                       "library" => %{
-                         "name" => "analytics_elixir",
-                         "transport" => "http",
-                         "version" => ^version
-                       },
-                       "location" => nil,
-                       "os" => nil,
-                       "page" => nil,
-                       "referrer" => nil,
-                       "screen" => nil,
-                       "timezone" => nil,
-                       "traits" => nil,
-                       "userAgent" => nil
-                     },
-                     "event" => "test1",
-                     "properties" => %{},
-                     "timestamp" => nil,
-                     "type" => "track",
-                     "userId" => nil
-                   }
-                 ]
-               } = Poison.decode!(received_body)
-
         # messageId and sentAt are not asserted
+        %{"batch" => [received_event | _received_events]} =
+          received_body
+          |> Poison.decode!()
+          |> Map.delete("sentAt")
 
-        Plug.Conn.resp(conn, 200, "")
-      end)
+        received_event = Map.delete(received_event, "messageId")
 
-      event = %Segment.Analytics.Track{
-        userId: nil,
-        event: "test1",
-        properties: %{},
-        context: %Segment.Analytics.Context{}
-      }
+        assert %{"batch" => [received_event]} == expected_request_body
```

Comment:
Shouldn't we capture the log in the tests?

Reply:
Done!